### PR TITLE
Generalize UUID branch

### PR DESCRIPTION
**Pull request by maxsibilla**
Modifying UUID-API to be more generalized with both SenNet and HuBMAP in addition to allowing support for setting various types via configuration settings.

### CODE REVIEW

**Review by shirey — APPROVED** [review]

File: src/app.py [hunk 1]
```diff
@@ -55,10 +68,10 @@ def init():
         print("Check the log file for further information: " + LOG_FILE_NAME)
 
 
-
-@app.route('/', methods = ['GET'])
+@app.route('/', methods=['GET'])
 def index():
-    return "Hello! This is HuBMAP UUID API service :)"
+    return "Hello! This is the " + group_name + " UUID API service :)"
```
Comment (yuanzhou) [inline src/app.py:73]: Need None check here for `group_name` because you had the default value as `None`

File: src/app.py [hunk 2]
```diff
@@ -167,32 +183,34 @@ def get_hmuuid(hmuuid):
...
-def is_hmuuid(hmuuid):
+            return Response("Invalid request use GET to retrieve UUID information", 500)
+    except Exception as e:
+        eMsg = str(e)
+        logger.error(e, exc_info=True)
+        return (Response("Unexpected error: " + eMsg, 500))
+
+
+@app.route('/' + app_uuid + '/<uuid>/exists', methods=["GET"])
+@secured(groups=group_name + "-read")
```
Comment (yuanzhou) [inline src/app.py:194]: @shirey @maxsibilla with the use of AWS API Gateway, do we still need this `@secured` decorator?

File: src/app.py [hunk 3]
```diff
@@ -167,32 +183,34 @@ def get_hmuuid(hmuuid):
...
-@secured(groups="HuBMAP-read")
-def is_hmuuid(hmuuid):
+            return Response("Invalid request use GET to retrieve UUID information", 500)
+    except Exception as e:
+        eMsg = str(e)
+        logger.error(e, exc_info=True)
+        return (Response("Unexpected error: " + eMsg, 500))
+
+
+@app.route('/' + app_uuid + '/<uuid>/exists', methods=["GET"])
```
Comment (yuanzhou) [inline src/app.py:193]: We probably can use an API key through AWS API Gateway...

**Review by yuanzhou — COMMENTED** [review]
More discussions may need to move forward

File: src/app.py (hunk 2 above)
Comment (shirey) [reply src/app.py:194]: @yuanzhou Leave the secured decorators for now.  Once we've got the new gateway stuff 100% in place we can remove most (all?) of these.

File: src/app.py (hunk 1 above)
Comment (maxsibilla) [reply src/app.py:76]: I now raise an exception in both app.py and uuid_worker when setting these values from the config file if "API_TYPE" is absent

**Pull request merged**